### PR TITLE
Add B&C test reporting

### PR DESCRIPTION
**Pull request by toban**
- Adds a JsonReporter that logs the test runs to a file by suite and stores it in TestArtifacts
- Adds a reporter step that brings failing tests up to the github UI

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        suite: [ repo, fedprops, repo-client ]
+        suite: [ repo, fedprops ]
```
Comment (tarrow) [inline .github/workflows/test.yml:20]: Is this supposed to be included in this commit?

Comment (toban) [reply .github/workflows/test.yml:20]: no that looks wrong

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        suite: [ repo, fedprops, repo-client ]
```
Comment (mariushoch) [inline .github/workflows/test.yml:20]: Seems like an accidental revert

File: test/test_reporter.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+set -e
+
+cd test
+
+docker-compose \
+    -f docker-compose-selenium-test.yml \
+    build
+docker-compose \
+    -f docker-compose-selenium-test.yml \
+    run --rm wikibase-selenium-test npm run "test:reporter"
```
Comment (mariushoch) [inline test/test_reporter.sh:11]: Where is this (`test:reporter`) defined?

**Review by mariushoch — CHANGES_REQUESTED** [review]
Looks good in general, but didn't test.

File: test/test_reporter.sh (hunk 3 above)
Comment (toban) [reply test/test_reporter.sh]: this can go!

**Comment by mariushoch** [comment]
Maybe `test/test_suite.sh` should remove `./test/log/selenium/result-$SUITE.json` (or `onRunnerEnd` doesn't need to amend)? Otherwise the logs just accumulate over time.

**Pull request merged**